### PR TITLE
Move common variables out of cache test blocks

### PR DESCRIPTION
**Pull request by derobins**
This function has stack size warnings and needs some fixin'

### CODE REVIEW

File: test/cache.c [hunk 1]
```diff
@@ -4145,6 +4145,16 @@ check_flush_cache__flush_ops(H5F_t *file_ptr)
...
 
+    /* Per-test variables. Each test sets these. */
+    int          test_num;
+    unsigned int flush_flags;
+    int          spec_size;
+    int          check_size;
+    unsigned     init_expected_index_len;
+    size_t       init_expected_index_size;
+    unsigned     expected_index_len;
+    size_t       expected_index_size;
```
Comment (derobins) [inline test/cache.c:4156]: There are so many tests w/ repeated block variables in this function, it generates a stack size warning w/ XCode's clang

**Comment by derobins** [comment]
This still has stack size warnings w/ clang. I'm going to have to separate the giant function into separate tests :(

**Review by byrnHDF — APPROVED** [review]

**Comment by jhendersonHDF** [comment]
> This still has stack size warnings w/ clang. I'm going to have to separate the giant function into separate tests :(

Are Clang and GCC using the same flags? I see the `stack-usage=8192` flag for GCC, but doesn't look like we set anything for Clang.

**Review by jhendersonHDF — APPROVED** [review]

**Review by qkoziol — APPROVED** [review]

**Comment by derobins** [comment]
> > This still has stack size warnings w/ clang. I'm going to have to separate the giant function into separate tests :(
> 
> Are Clang and GCC using the same flags? I see the `stack-usage=8192` flag for GCC, but doesn't look like we set anything for Clang.

We should probably do a pass to see if they are in sync. I noticed it on MacOS.

**Pull request merged**